crates/pedersen/Cargo.toml

```diff
@@ -23,11 +19,14 @@ ff = { git = "https://github.com/eqlabs/ff", branch = "derive_bitvec", default-f
     "derive",
     "alloc",
 ] }
+serde = "1.0.134"
```

Comment:
I think pretty much all of this can just be replaced with a `to_hex_str` method?

Reply:
In such case we would have to drop the Serialize & Deserialize impls for StarkHash.
As a consequence we wouldn't be able to derive `Deserialize, Serialize` and have newtype serde for free (this saves a lot of boilerplate.)

Reply:
Btw to_hex_str would either have to use heap or accept mut slice to avoid it.

Reply:
So its either +1 dependency or boilerplate.

Reply:
I'm fairly sure serde would have to heap allocate, but either way we don't really worry about heap allocation..

Reply:
Actually we really don't have to heap allocate beyond whatever serde does under the hood, so no worries wrt heap allocation. There's a tradeoff here: keep `StarkHash::serialize` or increase boilerplate and use field annotations instead?